spec/lib/license_finder/package_managers/go_dep_spec.rb

```diff
@@ -156,10 +152,105 @@ module LicenseFinder
+            end
+
+            after do
+              ENV['GOPATH'] = @orig_gopath
+            end
+
+            it 'should return an array of packages' do
+              packages = subject.current_packages
+              expect(packages.map(&:name)).to include('github.com/pivotal/foo', 'github.com/pivotal/bar')
+              expect(packages.map(&:version)).to include('61164e4', '3245708')
+            end
+
+            it 'should set the install_path to the GOPATH' do
+              packages = subject.current_packages
+              expect(packages[0].install_path).to eq('/fake/go/path/src/github.com/pivotal/foo')
+              expect(packages[1].install_path).to eq('/fake/go/path/src/github.com/pivotal/bar')
+            end
+          end
+
+          context 'when GOPATH is not set' do
+            before do
+              @orig_gopath = ENV['GOPATH']
+              @orig_home = ENV['HOME']
+              ENV.delete('GOPATH')
+              ENV['HOME'] = '/some-home'
+
+              allow(FileTest).to receive(:directory?).with("/fake/path/#{GoDep::OLD_GODEP_VENDOR_PATH}").and_return(false)
+              allow(FileTest).to receive(:directory?).with("/fake/path/#{GoDep::GODEP_VENDOR_PATH}").and_return(false)
+
+              expect(SharedHelpers::Cmd).to receive(:run).with('godep restore').and_return(['output', nil, cmd_success])
+            end
+
+            after do
+              ENV['GOPATH'] = @orig_gopath
+              ENV['HOME'] = @orig_home
+            end
+
+            it 'should return an array of packages' do
+              packages = subject.current_packages
+              expect(packages.map(&:name)).to include('github.com/pivotal/foo', 'github.com/pivotal/bar')
+              expect(packages.map(&:version)).to include('61164e4', '3245708')
+            end
+
+            it 'should set the install_path to the GOPATH' do
+              packages = subject.current_packages
+              expect(packages[0].install_path).to eq('/some-home/go/src/github.com/pivotal/foo')
+              expect(packages[1].install_path).to eq('/some-home/go/src/github.com/pivotal/bar')
+            end
+          end
+
+          context 'when godep restore fails' do
+            let(:failed_status) { double(Process::Status, success?: false, exitstatus: 0) }
+
+            before do
+              @orig_gopath = ENV['GOPATH']
+              ENV['GOPATH'] = '/fake/go/path'
+
+              allow(FileTest).to receive(:directory?).with("/fake/path/#{GoDep::OLD_GODEP_VENDOR_PATH}").and_return(false)
+              allow(FileTest).to receive(:directory?).with("/fake/path/#{GoDep::GODEP_VENDOR_PATH}").and_return(false)
+
+              expect(SharedHelpers::Cmd).to receive(:run).with('godep restore').and_return([nil, 'some-error-message', failed_status])
+            end
+
+            after do
+              ENV['GOPATH'] = @orig_gopath
+            end
+
+            it 'should raise error' do
```

Comment:
Is this going to be a hard stop? Meaning is LF suppose to quit if it cannot download the sources in this scenario?

Reply:
The cases we think the download command would fail will be:
- the project is broken
- randomly just failed
- network died in the middle

In those cases, I think it would be better to error out instead of returning partial data.

Also, we were going through gradle, and looks like gradle also fails out when gradle command fails.
So we think it should be a hard stop, so that we can keep consistency between different package managers.